File: 0409-longest-palindrome/0409-longest-palindrome.cpp
class Solution {
public:
    int longestPalindrome(string s) {
        vector<int> list1(128);
        int check = 0;
        int result = 0;
        for(char c : s) list1[c]++;
        
        for(int i : list1){
            if(i % 2 == 0) result += i;
            else{
                check = 1;
                result += (i - 1);
            }
        }
        if(check == 1) result += 1;
        return result;
    }
};

File: 0409-longest-palindrome/0409-longest-palindrome.py
class Solution(object):
    def longestPalindrome(self, s):
        """
        :type s: str
        :rtype: int
        """
        list1 = [0]*256
        result = 0
        check = 0
        for i in range(len(s)): 
            list1[ord(s[i])] += 1
        for i in range(len(list1)): 
            if list1[i] % 2 == 0: 
                result += list1[i]
            else: 
                check = 1
                result += (list1[i]-1)
        if check == 1: 
            result += 1
        return result